x-pack/plugins/security_solution/public/management/cypress/e2e/policy/policy_details.cy.ts

```diff
@@ -104,17 +114,28 @@ describe(
         });
 
         it('should successfully update the manifest version to custom date', () => {
+          const todayMinusTwoDays = moment.utc().subtract(2, 'days');
+
           loadProtectionUpdatesUrl(policy.id);
-          cy.getByTestSubj('protectionUpdatesSaveButton').should('be.disabled');
-          cy.getByTestSubj('protection-updates-manifest-switch').click();
+          expectSavedButtonToBeDisabled();
+          clickAutomaticUpdatesToggle();
           cy.getByTestSubj('protection-updates-manifest-note').type(testNote);
 
-          cy.intercept('PUT', `/api/fleet/package_policies/${policy.id}`).as('policy');
+          cy.intercept('PUT', `/api/fleet/package_policies/${policy.id}`, (req) => {
+            // Parse the date from the request and subtract one day
```

Comment:
why do you modify the request here? to avoid interacting with the date picker? if so, I'd suggest to rather interact with the date picker : ) this looks quite cryptic, and also would be better to test what the user can do

Reply:
It would be better to test what users can actually do ;), however:
1. The datepicker doesn't expose selectors that would make it easy and reliable to select dates. We could work around this by introducing logic to select dates manually, but this might introduce edge cases that are hard to anticipate and would make the test flaky sooner or later.
2. Since the field is an input, we could manually set its value. However, we would need to click outside the element to make the change stick and trigger a UI update. This seems a bit hacky, and such actions have caused flakiness in the past.

I decided to go with the API request to make the test as stable as possible. I would prefer to test the full user journey but not at the cost of introducing flakiness. In the end, the datepicker is an external component already being tested by another team.

We can discuss this further if you strongly disagree :)

Reply:
oh, I see your point.
see, that's why FTR tests can be better, because there are a tons of 'page objects' you can use to do the dirty work instead of you :D

agree with your arguments, but there may be no need to click anywhere on the picker - in the end it's just an input field which gets a supporting date picker in a portal if it's active, but it also works if you just enter/paste a text in the correct format (e.g. `May 16, 2024`)

if this is easy to do (didn't test it in cy, just clicked around in a local instance), I'd suggest to change to this, but otherwise I agree with your arguments, but would be happy to see some comments explaining why we need the interception

Reply:
Okay, you convinced me :D. Im going to try to set this date in an input and see if it's stable in the CI.